### PR TITLE
Fix race condition in e2e tests

### PR DESCRIPTION
**Pull request by erbesharat**
**What this PR does / why we need it**:
Creates a channel for the DNS server to kill the go routine when the test is finished.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by stp-ip** [comment]
Did a first look and am still thinking about a solution.


CoreDNS does use a test server too, so I wanted to leave this here already:
https://github.com/coredns/coredns/blob/master/plugin/test/server.go

Example test:
https://github.com/coredns/coredns/blob/master/plugin/file/lookup_test.go

**Comment by stp-ip** [comment]
Looks like a good start. Let's discuss this in Basecamp to get a clear handle of the e2e tests we want.

File: txtdirect_test.go [hunk 1]
```diff
@@ -29,6 +30,12 @@ import (
 
 // Testing TXT records
 var txts = map[string]string{
+	// type=host
+	"_redirect.e2e.txtdirect.":       "v=txtv0;to=https://e2e.txtdirect.org;type=host",
```
[the remark this answers is not in the document]
Comment (erbesharat) [reply txtdirect_test.go]: ``.txtdirect`` is the TLD in those URLs. We can change it to something else in the call to ``dns.HandleFunc()``

[the remark this answers is not in the document]
Comment (stp-ip) [reply txtdirect_test.go]: Ok then it makes sense. Just wanted to check the intention.

**Pull request merged**